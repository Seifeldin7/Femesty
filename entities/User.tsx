export default class User {
    public id!: string;
    public email: string;
    public username: string;
    public name: string;
    public image: string;
    public description: string;
    public phone: string;
    public address: string;
    public created_at: string;
    public password_reset_pin!: string;

    constructor(email: string,
        created_at: string,
        username: string,
        name: string,
        image: string,
        description: string,
        phone: string,
        address: string) {
        this.email = email;
        this.created_at = created_at;
        this.address = address;
        this.description = description;
        this.image = image;
        this.name = name;
        this.phone = phone;
        this.username = username;
    }
}
