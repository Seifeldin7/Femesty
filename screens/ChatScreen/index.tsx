import { Ionicons } from "@expo/vector-icons";
import React, { Component } from "react";
import { View, StatusBar, Text, TouchableOpacity } from "react-native";
import { FlatList } from "react-native-gesture-handler";
import { connect } from "react-redux";
import { ChatInputBox } from "../../components/ChatScreenComponents/ChatInputBox";
import { ChatMessage } from "../../components/ChatScreenComponents";
import { styles } from "./styles";
import { mapDispatchToProps, mapStateToProps, Props, State } from "./types";
import { ScreenSkeleton } from "../../components/ThemeComponents/ScreenSkeleton";

export class NCChatScreen extends Component<Props, State> {
  constructor(props: Readonly<Props>) {
    super(props);

    this.state = {};
    this.sendMessage = this.sendMessage.bind(this);
  }

  sendMessage = async (message: any) => {
    try {
      await this.props.AddMessage(message);
    } catch (err) {
      console.log(err);
    }
  };

  async componentDidMount() {
    try {
      await this.props.FetchMessages();
    } catch (err) {
      console.log(err);
    }
  }

  render() {
    return (
      <ScreenSkeleton
        goBackNavigation={() => this.props.navigation.goBack()}
        title={"Live Chat"}
      >
        <FlatList
          data={this.props.messages.messages}
          renderItem={({ item }) => <ChatMessage message={item} />}
          keyExtractor={(item: any) => item.id.toString()}
          inverted
        />
        <ChatInputBox
          sendMessageFunction={(message: any) => this.sendMessage(message)}
        />
      </ScreenSkeleton>
    );
  }
}

export const ChatScreen = connect(
  mapStateToProps,
  mapDispatchToProps
)(NCChatScreen);
