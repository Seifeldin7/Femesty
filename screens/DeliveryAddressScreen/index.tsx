import React, { Component } from "react";
import { View } from "react-native";
import { styles } from "./styles";
import { connect } from "react-redux";
import { mapDispatchToProps, mapStateToProps, Props, State } from "./types";
import { Form } from "../../components/SharedComponents/Form";
import { Button } from "../../components/SharedComponents/Buttons";
import { BasicColors, MainColors } from "../../styles/Colors";
import { ButtonStyles } from "../../styles/Buttons";
import { ScreenSkeleton } from "../../components/ThemeComponents/ScreenSkeleton";
import { DoubleVz21Container } from "../../styles/Containers";

export class NCDeliveryAddressScreen extends Component<Props, State>
{
    constructor(props: Readonly<Props>) {
        super(props);
        this.state = {
            submitIsPressed: false,
            formValues: {}
        };
        this.onConfirm = this.onConfirm.bind(this);
        this.updateAddress = this.updateAddress.bind(this);
    }
    async onConfirm(formValues: any) {
        await this.props.SENDCODE(formValues.phone);
        this.setState({ formValues: formValues });
        this.props.navigation.navigate("PhoneConfirmation", {
            verify: this.props.VERIFYCODE,
            updateAddress: this.updateAddress, 
            cancelcode: this.props.CANCELCODE
        });
    }
    async updateAddress() {
        let userAddress = this.state.formValues;
        this.props.EDITADDRESS(userAddress.name, userAddress.phone, userAddress.address);
        this.props.navigation.navigate("Profile");
    }
    render() {
        return (
            <ScreenSkeleton
                goBackNavigation={() => { this.props.navigation.navigate("Profile") }}
                title={"Delivery Details"}
                rightControlButtonNavigation={() =>
                    this.props.navigation.navigate("Profile")
                }
            >
                <View style={DoubleVz21Container.Container}>
                    <View style={{ ...DoubleVz21Container.TopComponent, ...styles.row2 }}>
                        <Form
                            submitIsPressed={this.state.submitIsPressed}
                            onSubmit={this.onConfirm}
                            formData={{
                                name: this.props.profile.profile.name,
                                phone: this.props.profile.profile.phone,
                                address: this.props.profile.profile.address

                            }}
                            inputProps={{
                                name: {
                                    label: "Name",
                                    placeholder: "Jane Doe",
                                },
                                phone: {
                                    label: "Phone Number",
                                    placeholder: "+20213456789",
                                    autoCompleteType: "tel",
                                    errorText: "Please enter a valid phone number!"

                                },
                                address: {
                                    label: "Address",
                                    placeholder: "13th street Egypt square",
                                }
                            }} />
                    </View>
                    <View style={{ ...DoubleVz21Container.BottomComponent, ...styles.row3 }}>
                        <Button
                            style={{ ...ButtonStyles.Large, alignSelf: 'center' }}
                            textColor={BasicColors.White}
                            backgroundColor={MainColors.Light}
                            enable={true}
                            title="Confirm"
                            onPress={() => this.setState({ submitIsPressed: true })} />
                    </View>
                </View>
            </ScreenSkeleton>
        );
    }
}

export const DeliveryAddressScreen = connect(mapStateToProps, mapDispatchToProps)(NCDeliveryAddressScreen);
