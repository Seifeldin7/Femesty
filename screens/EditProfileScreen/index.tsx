import React, { Component } from "react";
import {
  View,
  Text,
  TouchableOpacity,
} from "react-native";
import { connect } from "react-redux";
import { styles } from "./styles";
import { mapDispatchToProps, mapStateToProps, Props, State } from "./types";
import { Form } from "../../components/SharedComponents/Form";
import { ScreenSkeleton } from "../../components/ThemeComponents/ScreenSkeleton";
import { TripleVz123Container } from "../../styles/Containers";
import { Avatar } from "../../components/ThemeComponents/Avatar";

export class NCEditProfileScreen extends Component<Props, State> {
  constructor(props: Readonly<Props>) {
    super(props);

    this.state = {
      submitIsPressed: false
    };
    this.saveChanges = this.saveChanges.bind(this);
  }

  async saveChanges(formValues: any) {
    await this.props.EditProfile(formValues.name, formValues.username);
    this.props.navigation.goBack();
  }

  render() {
    return (
      <ScreenSkeleton
        goBackNavigation={() => this.props.navigation.goBack()}
        title={"Edit Profile"}
      >
        <View style={TripleVz123Container.Container}>
          <View style={TripleVz123Container.TopComponent}>
            <TouchableOpacity
              style={styles.saveView}
              onPress={() => {
                this.setState({ submitIsPressed: true })
                this.props.navigation.goBack();
              }}
            >
              <Text style={styles.save}>Save Changes</Text>
            </TouchableOpacity>
          </View>
          <View style={[TripleVz123Container.MiddleComponent, { alignSelf: 'center' }]}>
            {this.props.profile.profile.image ? (
              <Avatar imgPath={{ uri: this.props.profile.profile.image }}
                size="large" borderRadius="rounded-rectangle" />
            ) : (
              <Avatar imgPath={require("../../assets/images/default_profile_pic.jpg")}
                size="large" borderRadius="rounded-rectangle" />
            )}
          </View>
          <View style={[TripleVz123Container.BottomComponent, styles.inputs]}>
            <Form
              submitIsPressed={this.state.submitIsPressed}
              onSubmit={this.saveChanges}
              formData={{
                name: this.props.profile.profile.name,
                username: this.props.profile.profile.username,
              }}
              inputProps={{
                name: {
                  label: "Name",
                  placeholder: "@name",
                },
                username: {
                  label: "User Name",
                  placeholder: "@username",

                },
              }} />
          </View>
        </View>
      </ScreenSkeleton>
    );
  }
}

export const EditProfileScreen = connect(
  mapStateToProps,
  mapDispatchToProps
)(NCEditProfileScreen);
