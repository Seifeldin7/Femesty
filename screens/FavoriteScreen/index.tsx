import React, { Component } from "react";
import {
  View,
  StatusBar,
  Text,
  TouchableOpacity,
  Button,
  Dimensions,
} from "react-native";
import { styles } from "./styles";
import { FavoriteHzScroller } from "../../components/FavoriteScreenComponents";

import { connect } from "react-redux";
import { mapDispatchToProps, mapStateToProps, Props, State } from "./types";
import { Ionicons } from "@expo/vector-icons";
import { checkConnected } from "../../utility/CheckConnection";
import { NoConnection } from "../../components/ThemeComponents/Views/NoConnectionView";
import { NoContent } from "../../components/ThemeComponents/Views/NoContentsView";
import { FavoritesScreenLoading } from "../../components/ThemeComponents/Views/LoadingViews";
import { ScreenSkeleton } from "../../components/ThemeComponents/ScreenSkeleton";

export class NCFavoriteScreen extends Component<Props, State> {
  constructor(props: Readonly<Props>) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    this.props
      .FetchFavorites()
      .then()
      .catch((err) => console.log(err));
  }

  CheckConnection = () => {
    let result = false;
    checkConnected()
      .then((res: boolean) => {
        result = res;
      })
      .catch((err) => { });
    return result;
  };

  render() {
    if (this.CheckConnection()) {
      return <NoConnection onCheck={this.props.FetchFavorites} />;
    } else if (this.props.favorites.loading) {
      return (
        <ScreenSkeleton
          goBackNavigation={() => this.props.navigation.goBack()}
          title={"My Favorites"}>

          <View style={{ flex: 1 }} />
          <View style={{ flex: 8, marginLeft: 40 }}>
            <FavoritesScreenLoading />
          </View>
          <View style={{ flex: 2.5 }} />
        </ScreenSkeleton>
      );
    } else if (!this.props.favorites.favorites.length) {
      return (
        <ScreenSkeleton
          goBackNavigation={() => this.props.navigation.goBack()}
          title={"My Favorites"}>

          <View style={{ flex: 1 }} />
          <View
            style={{
              flex: 12,
              marginRight: Dimensions.get("window").width / 15,
            }}
          >
            <NoContent
              onCheck={this.props.FetchFavorites}
              name={"Favorites"}
              description={
                "Nothing on the Favorites right now, Get started by liking some Looks."
              }
            />
          </View>
          <View style={{ flex: 2.5 }} />
        </ScreenSkeleton>

      );
    } else {
      return (
        <ScreenSkeleton
          goBackNavigation={() => this.props.navigation.goBack()}
          title={"My Favorites"}>

          <View style={{ flex: 1 }} />
          <View style={{ flex: 8, marginLeft: 40 }}>
            <FavoriteHzScroller
              favorites={this.props.favorites.favorites}
              deleteFavorite={this.props.DeleteFavorite}
            />
          </View>
          <View style={{ flex: 2.5 }} />
        </ScreenSkeleton>
      );
    }
  }
}

export const FavoriteScreen = connect(
  mapStateToProps,
  mapDispatchToProps
)(NCFavoriteScreen);
