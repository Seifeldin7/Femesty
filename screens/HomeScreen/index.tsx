import React, { Component } from "react";
import { View, StatusBar, TouchableOpacity } from "react-native";
import { styles } from "./styles";
import { Logo } from '../../components/ThemeComponents/Logo';
import { LooksResultVerticalScroller } from "../../components/SharedComponents/LooksResultVerticalScroller";
import { connect } from "react-redux";
import { mapDispatchToProps, mapStateToProps, Props, State } from "./types";
import { Icon } from "../../components/ThemeComponents/Icon";

export class NCHomeScreen extends Component<Props, State>
{
    constructor(props: Readonly<Props>) {
        super(props);

        this.state = {
        };
    }
    render() {
        return (
            <View style={styles.container} >
                <StatusBar
                    translucent
                    backgroundColor="transparent"
                    barStyle="light-content"
                />
                <View style={styles.row1}>
                    <TouchableOpacity activeOpacity={1.0} onPress={
                        () => { this.props.RefreshLooks(); this.props.FetchLooks(2) }
                    } style={styles.logo}>
                        <Logo size="medium" />
                    </TouchableOpacity>
                    <Icon
                        outerStyle={styles.Icon}
                        size="medium"
                        iconPath={require("../../assets/icons/small_icons/search.png")}
                        onPress={() => this.props.navigation.navigate("Search")} />
                </View>
                <LooksResultVerticalScroller
                    data={this.props.looks.looks}
                    getNextLook={(count: number) => { return this.props.FetchLooks(count); }}
                    refreshLooks={() => { return this.props.RefreshLooks() }}
                    addFavorite={this.props.AddFavorite}
                    deleteFavorite={this.props.DeleteFavorite}
                    navigation={this.props.navigation} />
            </View>
        );
    }

}

export const HomeScreen = connect(mapStateToProps, mapDispatchToProps)(NCHomeScreen);
