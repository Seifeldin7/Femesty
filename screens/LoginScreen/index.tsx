import React, { Component } from "react";
import { View, Text } from "react-native";
import { SocialButton } from "../../components/SharedComponents/Buttons";
import { TermsAndConditions } from "../../components/LoginScreenComponents/TermsAndConditions";
import { Popup } from "../../components/SharedComponents/Popup";
import { Logo } from "../../components/ThemeComponents/Logo";

import { mapDispatchToProps, mapStateToProps, Props, State } from "./types";
import { connect } from "react-redux";

import { styles } from "./styles";
import {
  DoubleVzEqualContainer,
  TripleVz131Container,
} from "../../styles/Containers";
import { TranslucentStatusBar } from "../../components/ThemeComponents/StatusBars";

export class NCLoginScreen extends Component<Props, State> {
  async componentDidMount() {
    this.props.LogIn();
  }

  render() {
    return (
      <View style={styles.screen}>
        <TranslucentStatusBar />
        <View style={DoubleVzEqualContainer.Container}>
          <View style={styles.topContainer}>
            <View style={{
              height: '80%', alignItems: "center",
              justifyContent: "flex-end",
            }}>
              <Logo size="large" color="white" />
              <View>
                <Text style={styles.title}>FEMESTY</Text>
                <Text style={styles.slogan}>Shop the Look</Text>
              </View>
            </View>

          </View>
          <View style={DoubleVzEqualContainer.BottomComponent} />
          <Popup isVisible={true}>
            <View style={[TripleVz131Container.Container, {}]}>
              <View style={styles.popup}>
                <Text style={styles.header}>Sign Up For Femesty</Text>
              </View>
              <View
                style={[
                  TripleVz131Container.MiddleComponent,
                  { flexDirection: "row" },
                ]}
              >
                <View style={{ flex: 1 }} />
                <View style={{ flex: 20 }}>
                  <SocialButton
                    title="SIGN UP WITH APPLE"
                    icon="apple"
                    color="black"
                    onPress={() => this.props.AppleLogin()}
                  />
                  <SocialButton
                    title="SIGN UP WITH FACEBOOK"
                    icon="facebook"
                    color="#3b5998"
                    onPress={() => { this.props.FacebookLogin(); }}
                  />
                  <SocialButton
                    title="SIGN UP WITH GOOGLE"
                    icon="google"
                    color="#DB4437"
                    onPress={() => this.props.GoogleLogin()}
                  />
                </View>
                <View style={{ flex: 1 }} />
              </View>
              <View style={TripleVz131Container.BottomComponent}>
                <View style={styles.termsAndConditions}>
                  <TermsAndConditions />
                </View>
              </View>
            </View>
          </Popup>
        </View>
      </View>
    );
  }
}

export const LoginScreen = connect(
  mapStateToProps,
  mapDispatchToProps
)(NCLoginScreen);
