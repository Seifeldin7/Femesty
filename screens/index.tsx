export * from "./ChatScreen"
export * from "./DeliveryAddressScreen"
export * from "./EditProfileScreen"
export * from "./FavoriteScreen"
export * from './HomeScreen';
export * from "./LoginScreen";
export * from "./NotificationsScreen"
export * from "./OrderScreen";
export * from "./PhoneConfirmationScreen";
export * from "./ProfileScreen"
export * from "./SearchResultScreen"
export * from "./SearchScreen"
export * from "./SplashScreen"