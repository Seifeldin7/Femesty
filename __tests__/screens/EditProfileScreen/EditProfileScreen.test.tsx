import React from "react";
import { fireEvent, render } from "react-native-testing-library";
import { ReactTestInstance } from "react-test-renderer";
import User from "../../../entities/User";
import { NCEditProfileScreen } from "../../../screens";

jest.mock('react-native-gesture-handler');
jest.mock('../../../../services/ApiClient', () => ({
    constants: {
        manifest: {
            developer: {
                projectRoot: '/home/test/project',
            },
            logUrl: 'https://localhost:19001/logs',
            extra: {
                baseUrl: '192.168.1.1'
            }
        }
    },
}));

test('should render edit profile screen correct', () => {

    let stylist = new User("s.seif@yahoo.com",
        "11 Nov 2020",
        "ahmed",
        "Ahmed Mohamed",
        "",
        "Famous Designer",
        "01111111",
        "Mokattam");

    const props = {
        EditProfile: jest.fn(),
        navigation: {goBack: jest.fn(), reset: jest.fn(), navigate: jest.fn()},
        profile: {
            profile: stylist,
        }
    };
    const tree = render(
        <NCEditProfileScreen
            {...props}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();

});

describe('<Edit Profile>', () => {
    let getByText: (text: string) => ReactTestInstance | null;
    let editProfile = jest.fn();
    let back = jest.fn();
    let stylist = new User("s.seif@yahoo.com",
        "11 Nov 2020",
        "ahmed",
        "Ahmed Mohamed",
        "",
        "Famous Designer",
        "01111111",
        "Mokattam");
    const props = {
        EditProfile: editProfile,
        navigation: {goBack: back, reset: jest.fn(), navigate: jest.fn()},
        profile: {
            profile: stylist,
        }
    };
    beforeEach(() => {
        ({ getByText } = render(
            <NCEditProfileScreen
            {...props}
        />
        ))
    });
    test('on Save', () => {
        const save = getByText("Save Changes");
        expect(save).toBeDefined();
        if (save) {
            fireEvent.press(save);
            expect(editProfile).toHaveBeenCalledTimes(1);
            expect(back).toHaveBeenCalledTimes(1);
        }
        
    });

});
