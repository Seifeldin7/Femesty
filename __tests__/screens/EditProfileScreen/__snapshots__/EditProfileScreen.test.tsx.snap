// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render edit profile screen correct 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "marginTop": null,
    }
  }
  horizontal={false}
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginLeft": 10,
            "opacity": 1,
            "paddingTop": 7,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "flex": 2.5,
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
            }
          }
        >
          Edit Profile
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 0.5,
          "marginHorizontal": 10,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#65CE88",
            "fontSize": 17,
          }
        }
      >
        Save Changes
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 20,
            "height": 120,
            "width": 120,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 3.5,
          "marginLeft": 15,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 21,
            "marginVertical": 8,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          Object {
            "fontSize": 21,
            "marginVertical": 8,
          }
        }
      >
        User Name
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
